apps/designer/app/canvas/features/text-editor/toolbar-connector.tsx

```diff
@@ -82,38 +82,58 @@ const $isSelectedLink = (selection: RangeSelection) => {
 
 export const ToolbarConnectorPlugin = () => {
   const [editor] = useLexicalComposerContext();
+  const [hasSelection, setHasSelection] = useState(false);
+
+  useEffect(() => {
+    if (hasSelection) {
```

Comment:
please use early return, it's hard to read in a review where that block scope ends